helpers/labgrid-raw-interface

```diff
@@ -75,22 +125,58 @@ def main(program, ifname, count):
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
-    parser.add_argument(
-        '-d',
-        '--debug',
-        action='store_true',
-        default=False,
-        help="enable debug mode"
+    parser.add_argument("-d", "--debug", action="store_true", default=False, help="enable debug mode")
+    subparsers = parser.add_subparsers(dest="program", help="program to run")
+
+    # tcpdump
+    tcpdump_parser = subparsers.add_parser("tcpdump")
+    tcpdump_parser.add_argument("ifname", type=str, help="interface name")
+    tcpdump_parser.add_argument("count", type=int, default=None, help="amount of frames to capture while recording")
+    tcpdump_parser.add_argument(
+        "--timeout", type=int, default=None, help="Amount of time to capture while recording. 0 means capture forever"
     )
-    parser.add_argument('program', type=str, help='program to run, either tcpreplay or tcpdump')
-    parser.add_argument('interface', type=str, help='interface name')
-    parser.add_argument('count', nargs="?", type=int, default=None, help='amount of frames to capture while recording')
+
+    # tcpreplay
+    tcpreplay_parser = subparsers.add_parser("tcpreplay")
+    tcpreplay_parser.add_argument("ifname", type=str, help="interface name")
+
+    # ip
+    ip_parser = subparsers.add_parser("ip")
+    ip_parser.add_argument("ifname", type=str, help="interface name")
+    ip_parser.add_argument("action", type=str, choices=["up", "down"], help="action, one of {%(choices)s}")
```

Comment:
This looks like it would be hard to extend with more options later.

Reply:
What should I do instead? `--up`/`--down`?

Reply:
Hmm, leave it as is. It can still be changed to a subparser or something custom if needed.